FGEGraphics/UISystem/UIClickableElement.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using FGEGraphics.UISystem;
+
+namespace FGEGraphics.UISystem;
+
+/// <summary>Represents a clickable UI element on the screen.</summary>
+public abstract class UIClickableElement : UIElement
+{
+    /// <summary>Ran when this element is clicked.</summary>
+    public event Action Clicked;
+
+    /// <summary>Whether the mouse is hovering over this element.</summary>
+    public bool Hovered = false;
+
+    /// <summary>Whether this element is being clicked.</summary>
+    public bool Pressed = false;
+
+    /// <summary>Data internal to a <see cref="UIClickableElement"/> instance.</summary>
+    public struct InternalData()
+    {
+        /// <summary>Whether this element can be interacted with.</summary>
+        public bool Enabled = true;
+    }
+
+    /// <summary>Gets or sets whether this element can be interacted with.</summary>
+    public bool Enabled
+    {
+        get => Internal.Enabled;
+        set
+        {
+            Internal.Enabled = value;
+            if (!value)
+            {
+                Hovered = false;
+                Pressed = false;
+            }
+        }
+    }
+
+    /// <summary>Data internal to a <see cref="UIClickableElement"/> instance.</summary>
+    public InternalData Internal = new();
+
+    /// <summary>Constructs the clickable element.</summary>
+    /// <param name="pos">The position of the element.</param>
+    /// <param name="onClick">Ran when the element is clicked.</param>
+    public UIClickableElement(UIPositionHelper pos, Action onClick = null) : base(pos)
+    {
+        if (onClick is not null)
+        {
+            Clicked += onClick;
```

Comment:
This whole block can just be `Clicked = onClick` ... at which point the whole constructor can be removed too I think